packages/grid/_modules_/grid/components/GridRow.tsx

```diff
@@ -15,11 +15,37 @@ import { GridStateColDef } from '../models/colDef/gridColDef';
 import { GridCellIdentifier } from '../hooks/features/focus/gridFocusState';
 import { gridColumnsMetaSelector } from '../hooks/features/columns/gridColumnsSelector';
 import { useGridSelector } from '../hooks/utils/useGridSelector';
+import { GridRenderEditCellParams } from '../models';
+
+export interface GridRowIndexes {
+  /**
+   * Index of the row in the whole sorted and filtered dataset.
+   */
+  visibleRows: number;
+
+  /**
+   * Index of the row in the current page.
+   * If the pagination is disabled, this value will be equal to the `dataset` value.
+   */
+  pageRows: number;
+
+  /**
+   * Index of the row in the list of rows currently rendered by the virtualization engine.
+   * If the pagination is disabled, this value will be equal to the `page` value.
+   */
+  virtualizationEngineRows: number;
+}
 
-export interface GridRowProps {
+export interface GridRowProps extends React.HTMLAttributes<HTMLDivElement> {
   rowId: GridRowId;
   selected: boolean;
+  /**
+   * Index of the row in the whole sorted and filtered dataset.
+   * If some rows above have expanded children, this index also take those children into account.
+   * @deprecated Use `props.indexes.visibleRows` instead.
```

Comment:
I'm not sure about deprecating it. I don't see a use case for the other values passed in `indexes`. I would first fix `index` to pass the correct value.

Reply:
What is the correct value in your mind ?

Reply:
https://github.com/mui/mui-x/pull/3882#discussion_r800605561

Reply:
Formula fixed

Reply:
How about passing the index relative to only the rows reacheable by scroll (being `nextRenderContext.firstRowIndex! + i`) to `getRowClassName`? This way, developers can assign a custom class name without having to provide a custom row component. I don't like passing the `indexes` prop because if want to remove it later it will require a breaking change.

Reply:
I agree that passing the indexes to `getRowClassName` makes the example easier. I will improve the PR that way :+1: 

We could rename `useCurrentPageRows` => `useGridCurrentPageRows` to match the other exporter hooks and in that case I agree that user can handle there more specific usecases without the `indexes` prop.

Reply:
> We could rename useCurrentPageRows => useGridCurrentPageRows to match the other exporter hooks and in that case I agree that user can handle there more specific usecases without the indexes prop.

Sure, lets rename `useCurrentPageRows` and export it, in another PR maybe. I want to avoid deprecating props whenever is possible because in v6 it will be one more thing for developers to care about.

Reply:
If we do the export in another PR, what remains for this one ?

Reply:
What remains is to pass the indexes to `getRowClassName` and not deprecate the `index` prop. With that, developers can already create stripped rows without overriding the row component.

Reply:
I will finish this PR after https://github.com/mui/mui-x/pull/3848 is merged to avoid having two PRs adding the `GridRowClassNameParams`

Reply:
Yes, this PR is not urgent so I prefer to finish #3896 before :+1: